codex/stores/repostore.nim

```diff
@@ -144,18 +144,22 @@ method ensureExpiry*(
   if expiry <= 0:
     return failure(newException(ValueError, "Expiry timestamp must be larger then zero"))
 
-  let expiryKey = createBlockExpirationMetadataKey(cid)
-  let currentExpiry = await self.metaDs.get(expiryKey.tryGet)
+  without expiryKey =? createBlockExpirationMetadataKey(cid), err:
+    return failure(err)
 
-  if currentExpiry.isOk and expiry < (!currentExpiry).toSecondsSince1970:
-    trace "Current expiry is larger then the specified one, no action needed"
-    return success()
+  without currentExpiry =? await self.metaDs.get(expiryKey), err:
+    if err of DatastoreKeyNotFound:
+      error "No current expiry exists for the block"
+    else:
+      error "Could not read datastore key", err = err.msg
 
-  without (key, data) =? self.getBlockExpirationEntry(cid, expiry), err:
-    trace "Unable to create updated expiration metadata key", err = err.msg
     return failure(err)
 
-  if err =? (await self.metaDs.put(key, data)).errorOption:
+  if expiry < currentExpiry.toSecondsSince1970:
+    trace "Current expiry is larger then the specified one, no action needed"
```

Comment:
A minior change:
```nim
  if expiry <= currentExpiry.toSecondsSince1970:
    trace "Current expiry not shorter then the specified one, no action needed"
```

Reply:
Ok, btw. do you know the "Suggestion" feature for PRs? ;-) This would be great adept for that, for example...

https://haacked.com/archive/2019/06/03/suggested-changes/